lib/src/main/java/com/squareup/cycler/RecyclerMutations.kt

```diff
@@ -257,10 +342,76 @@ private class MutationExtension<T : Any>(val spec: MutationExtensionSpec<T>) : E
       lastActionState = actionState
     }
 
+    private fun removeSwipeState() {
+      swipeState = null
+    }
+
+    private fun createSwipeStateFor(dataItem: T, itemView: View) {
+      // TODO: particular underView specs could be added to
+      // main-recycler's row specs, to be used for those items only
+      // and keep things grouped by type of item (in the main recycler).
+      val underViewRowSpec = spec.swipeUnderViewSpecs.firstOrNull { it.matches(dataItem) }
+      underViewRowSpec?.createViewHolder(
+          CreatorContext(recycler), // recycler is not exactly correct, but well enoguh for now
+          subType = 0
+      )?.let { holder ->
+        val underView = holder.itemView
+        underView.measure(
+            MeasureSpec.makeMeasureSpec(itemView.width, MeasureSpec.EXACTLY),
+            MeasureSpec.makeMeasureSpec(itemView.height, MeasureSpec.EXACTLY)
+        )
+        underView.layout(0, 0, itemView.width, itemView.height)
+        swipeState = SwipeState(holder, SwipeBindData(dataItem, itemView))
+      }
+    }
+
+    /**
+     * This customizes drawing for swiping by drawing below the dragged view.
+     * We still call the standard method even for swiping as we want any standard processing
+     * (like translucency) to be applied.
+     */
+    override fun onChildDraw(
+      canvas: Canvas,
+      recyclerView: RecyclerView,
+      viewHolder: ViewHolder,
+      dX: Float,
+      dY: Float,
+      actionState: Int,
+      isCurrentlyActive: Boolean
+    ) {
+      if (actionState == ItemTouchHelper.ACTION_STATE_SWIPE) {
+        data.forPosition(
+            viewHolder.adapterPosition,
+            onDataItem = { dataItem ->
+              swipeState?.let { state ->
+                val width = viewHolder.itemView.width
+                val direction = if (dX < 0) ItemTouchHelper.START else ItemTouchHelper.END
+                val percentage = if (width <= 0) 0f else abs(dX / width).coerceAtMost(1f)
+                state.bind(direction, percentage)
+                val count = canvas.save()
+                canvas.translate(
+                    viewHolder.itemView.left.toFloat(),
+                    viewHolder.itemView.top.toFloat()
+                )
+                val underView = state.viewHolder.itemView
+                underView.draw(canvas)
+                canvas.restoreToCount(count)
+              }
+            },
+            onExtraItem = { 0 },
+            orElse = { 0 }
+        )
+      }
+      super.onChildDraw(canvas, recyclerView, viewHolder, dX, dY, actionState, isCurrentlyActive)
+    }
+
+
+
```

Comment:
Nit: extra white space (did you auto-format?)

Reply:
Strongly encourage you to enable ktlint on this project, it would catch stuff like this in CI.